### DISCUSSION TITLE
Modification from Mach tutorial wing

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#655] [General]
Dear profressor PingHe now I am trying to reduce mass and drag of wing by modifying the tutorial Mach tutorial wing. I have try to reduce mass by adding self.add_objective("mass", scaler=1.0)  but the optimizer tell me that ipopt can not handle 2 objective. I saw some one use DAFoam to publish paper they can reduce mass and drag at the same time. So could profressor suggest me how can I correctly define the objective of mass and drag.  ![image](https://github.com/mdolab/dafoam/assets/115110434/539bf924-d01c-40b3-859b-85393a70ba0b)

**Comment by friedenhe** [comment]
Check lines 138, 234-236 from [here](https://github.com/DAFoam/tutorials/blob/main/NACA0012_Airfoil/multipoint/runScript.py)

**Reply by Ryupuriwatt** [reply]
Thankyou very much. Now it works :)

**Comment by NeuroNexusAI** [comment]
Dear Professor PingHe,

I’m working on reducing mass and drag for the Mach tutorial wing, but IPOPT struggles with multiple objectives. I’ve added self.add_objective("mass", scaler=1.0), but it’s not working. I’ve seen DAFoam papers manage both objectives effectively. Any tips on setting up and optimizing for both mass and drag?

Thanks!
